### PR TITLE
63 convert and use ppm measurements

### PR DESCRIPTION
**Pull request by rstrange-scottlogic**
Switches the values used for surface pressure and temperature to use the interpolated values, and refactors use of the pollutant types in the forecast_data class.

### CODE REVIEW

File: air-quality-backend/src/air_quality/etl/forecast/forecast_data.py [hunk 1]
```diff
@@ -80,18 +110,77 @@ def __init__(self, single_level_data: xr.Dataset, multi_level_data: xr.Dataset):
...
+            interpolated_data = dataset[required_datum.value].interp(
+                latitude=xr.DataArray(latitudes, dims="latitude"),
+                longitude=xr.DataArray(longitudes, dims="longitude"),
+                valid_time=xr.DataArray(valid_times, dims="valid_time"),
+                method="linear",
+            )
+            (interpolated_data
+             .set_index(valid_time=["valid_time"])
+             .set_index(latitude=["latitude"])
+             .set_index(longitude=["longitude"]))
```
Comment (amehta-scottlogic) [inline air-quality-backend/src/air_quality/etl/forecast/forecast_data.py:157]: What does this actually do? Is it like a database index?

File: air-quality-backend/src/air_quality/etl/forecast/forecast_data.py [hunk 2]
```diff
@@ -80,18 +110,77 @@ def __init__(self, single_level_data: xr.Dataset, multi_level_data: xr.Dataset):
...
+            longitude = in_situ_measurement["location"]["coordinates"][0]
+            latitude = in_situ_measurement["location"]["coordinates"][1]
+            valid_time = in_situ_measurement["measurement_date"].timestamp()
+
+            if latitude not in latitudes:
+                latitudes.append(latitude)
+            if longitude not in longitudes:
+                longitudes.append(longitude)
+            if valid_time not in valid_times:
+                valid_times.append(valid_time)
```
Comment (amehta-scottlogic) [inline air-quality-backend/src/air_quality/etl/forecast/forecast_data.py:140]: I'm not sure about this. What would happen if we had these cities:
London, (90, 90)
Near London, (90, 45)

I think we'd end up with:
latitides: [90]
longitudes: [90, 45]

But the function below would expect the list sizes to be the same I think?

File: air-quality-backend/src/air_quality/etl/forecast/forecast_data.py (hunk 1 above)
Comment (rstrange-scottlogic) [reply air-quality-backend/src/air_quality/etl/forecast/forecast_data.py:157]: I think so, the interp creates a dataset without indicies, and without index's the .sel call just fails. I think it's an index to help with searching, and due to xarray being tuned for large datasets it refuses to search on anything unindexed (a bit of a guess)

File: air-quality-backend/src/air_quality/etl/forecast/forecast_data.py (hunk 2 above)
Comment (rstrange-scottlogic) [reply air-quality-backend/src/air_quality/etl/forecast/forecast_data.py:140]: Oh that's a good point, I'll update and add a unit test to cover as well

Comment (rstrange-scottlogic) [reply air-quality-backend/src/air_quality/etl/forecast/forecast_data.py:140]: Actually this one appears to be ok, as I'm already casting these on different dimensions when interpolating. I've added two unit tests to showcase it though, just give me a shout if you think it's still an issue.

**Review by amehta-scottlogic — APPROVED** [review]

**Review by amehta-scottlogic — APPROVED** [review]

**Pull request merged**